### ISSUE TITLE
Restore command for TV Studios with dodgy firmware

### ISSUE THREAD

**Issue by wolfling366** [bitfocus/companion-module-bmd-atem#198]
Given that the ATEM firmware is (since 7.5.2) unable to retain the Key Label and Media Pool information upon loss of power, we have to restore it every time we start up.

Blackmagic Design has already [made it clear](Https://forum.blackmagicdesign.com/viewtopic.php?f=4&t=110658) that they have no intention of updating this problem, as they are focused on the newer Constellation devices amd the like.
We use other devices, and would prefer not to use an outdated app on our PC.
We do, however, have a "Reset to default" button on our Stream Deck that would be ideal for this issue.

Is anyone able to help with a command to restore a particular XML file to the ATEM?

I wonder if it could be done directly via BitFocus, or if it'd have to be done via the ATEM Software Control app... but figure it *should* be able to be done directly.

Thanks in advance, not only for a possible solution, but also just for being willing to consider our issue.

Asher.

**Comment by Julusian** [comment]
Replicating the restore from the xml file is hard for us to do. The atem software does it by sending many many commands to set every property on the atem, kind of simulating what would happen if you were to do the restore by hand.
Which means that to do properly will require companion to be able to understand and control every property on the atem. This module isnt anywhere near complete enough for this yet, and to get everything implemented is a lot more time than I can spare.

The best solution I can think of for now is to build out a trigger or button to configure everything you need through actions, and let us know what additional actions are needed so that those can be implemented. It wont be a quick or easy thing to get everything from the xml implemented, but hopefully enough of it can be done

**Comment by Julusian** [comment]
Also, that is a terrible response from blackmagic. The TVS HD was a current model right up until the Constellation HD series was released. So to have no intention of fixing such a breaking bug in current models that were still being advertised and sold is ridiculous

**Comment by Gartom** [comment]
Actually, Blackmagic has _not_ stated that they _won't_ fix the firmware for TVS HD. I asked Greg to make a status update in the BMD forum thread and it seems that BMD considers the workaround to load a saved configuration xml as good enough. Maybe someone that knows why this does not work in the Companion setup should post a response in the BMD forum thread.

**Comment by wolfling366** [comment]
Thank you for your reply on the BMG thread, Julian. Your statement was 100%
spot-on, and my entire team agree with you.
If they were to *at least* have included XML loading in the SDK so you had
the tools to implement it, that would be great, but they just didn't bother
to do anything at all.
I'm simply disappointed in their stance on this issue.

And thank you for stepping into the fray with a voice of authority.

Regards,
Asher.

**Comment by Gartom** [comment]
Yes, thanks a lot @Julusian for your reply in the BMD thread. I agree with @wolfling366 that it was spot on.

In our volounteer setup, we have reduced the need of people to run a live service from 6-7 down to 2-3 thanks to a stable setup using Companion, where one is running live FOH sound separately. The volounteers can focus on the live production instead of knowing nitty gritty settings in ATEM SW Control or other specific software to control our cameras, media and streaming.

Thanks again @Julusian for your support!
